mismi-s3/src/Mismi/S3/Commands.hs

```diff
@@ -622,6 +631,26 @@ downloadWithRange a start end dest = withRetries 5 $ do
     Just () -> pure ()
     Nothing -> liftIO $ ioError (userError "downloadWithRange timeout")
 
+downloadRecursive :: Address -> FilePath -> EitherT DownloadError AWS ()
+downloadRecursive src dest = do
+  -- Check if the destination already exists and is not a directory.
+  es <- tryIO $ getFileStatus dest
+  case es of
+    Left _ -> pure ()
+    Right st -> unless (isDirectory st) . left $ DownloadDestinationNotDirectory dest
```

Comment:
I think `isDirectory` here will be `True` if `dest` is a directory *or* a symlink to a directory (due to using `getFileStatus` above rather than `getSymbolicLinkStatus`) - this seems reasonable, just checking it's intentional.

Reply:
It wasn't actually intentional and I think it should be `getSymbolicLinkStatus` instead, because it should allow download to a symlinked directory.

Reply:
> it should allow download to a symlinked directory.

Yep, this should be the current behaviour - the names are a bit confusing, but `getFileStatus` will deref the symlink whereas `getSymbolicLinkStatus` will not.

Reply:
(`getFileStatus` calls `stat(3)` and `getSymbolicLinkStatus` calls `lstat(3)`.)